### PR TITLE
changed the way attributes are set

### PR DESCRIPTION
**Pull request by ashleyschuett**
- updated containership.plugin
 - if provider is avaliable only run that file
 - if provider is not avaliable run all providers to see which works

### CODE REVIEW

File: lib/cloud-hints.js [hunk 1]
```diff
@@ -6,24 +6,38 @@ const fs = require('fs');
...
+        const provider_name = _.get(config, 'provider');
         const providers = fs.readdirSync(`${__dirname}/providers`);
         let cloud_hints;
 
-        return async.some(providers, (provider_name, fn) => {
-            const provider = require(`${__dirname}/providers/${provider_name}`);
-
-            return provider.is_true((err, is_true, attrs) => {
+        const get_provider_hints = (provider_name, fn) => {
+            const provider = require([__dirname, "providers", provider_name].join("/"));
```
Comment (nicktate) [inline lib/cloud-hints.js]: can we use template literal strings here

File: lib/cloud-hints.js [hunk 2]
```diff
@@ -6,24 +6,38 @@ const fs = require('fs');
...
                 if (is_true) {
                     cloud_hints = attrs;
                     return fn(null, true);
                 }
 
                 return fn(null, false);
             });
+        }
+
+        const set_attributes = () => {
```
Comment (nicktate) [inline lib/cloud-hints.js]: unused function

File: lib/cloud-hints.js [hunk 3]
```diff
@@ -6,24 +6,38 @@ const fs = require('fs');
...
+            return cb(err, cloud_hints);
+        }
+
+        const provider_file = `${provider_name}.js`;
+        if(provider_name && _.include(providers, provider_file)) {
```
Comment (nicktate) [inline lib/cloud-hints.js]: nit: newline before `if`

File: lib/cloud-hints.js (hunk 1 above)
Comment (normanjoyner) [inline lib/cloud-hints.js]: then here we can just do `const provider = providers[provider_name];`

File: lib/plugin.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ class ContainershipCloudHintsPlugin extends ContainershipPlugin {
     startLeader(host) {
         super.startLeader(host);
 
-        return cloud_hints.get_hints((err, hints) => {
+        return cloud_hints.get_hints(this.config.core, (err, hints) => {
```
Comment (normanjoyner) [inline lib/plugin.js]: rearrange requires such that local requires are first (top of file)

File: lib/cloud-hints.js (hunk 3 above)
Comment (normanjoyner) [inline lib/cloud-hints.js]: `if(provider_name && providers[provider_name])`

File: lib/cloud-hints.js [hunk 5]
```diff
@@ -6,24 +6,38 @@ const fs = require('fs');
...
 
                 return fn(null, false);
             });
+        }
+
+        const set_attributes = () => {
+            return cb(err, cloud_hints);
+        }
+
+        const provider_file = `${provider_name}.js`;
```
Comment (normanjoyner) [inline lib/cloud-hints.js]: remove

File: lib/cloud-hints.js [hunk 6]
```diff
@@ -6,24 +6,38 @@ const fs = require('fs');
...
         const providers = fs.readdirSync(`${__dirname}/providers`);
         let cloud_hints;
 
-        return async.some(providers, (provider_name, fn) => {
-            const provider = require(`${__dirname}/providers/${provider_name}`);
-
-            return provider.is_true((err, is_true, attrs) => {
+        const get_provider_hints = (provider_name, fn) => {
+            const provider = require([__dirname, "providers", provider_name].join("/"));
+            provider.is_true((err, is_true, attrs) => {
```
Comment (normanjoyner) [inline lib/cloud-hints.js]: rename `is_true` to something that makes more sense such as `matches` ... can also remove `is_true` callback param in favor of passing everything through `err`

**Comment by nicktate** [comment]
can you add the lightsail provider

**Pull request merged**